library/src/main/scala/g8.scala

```diff
@@ -462,35 +474,45 @@ object G8 {
     }
   }
 
-  def readProps(stm: java.io.InputStream): G8.OrderedProperties = {
-    val p = new LinkedListProperties
-    p.load(stm)
-    stm.close()
-    (OrderedProperties.empty /: p.keyList) { (l, k) =>
-      l :+ (k -> p.getProperty(k))
+  def readProps(stm: InputStream): G8.OrderedProperties = {
+    // Overrides java.util.Properties to return properties back in the order they were specified
+    val properties = new Properties {
+      val parameters = mutable.MutableList.empty[(String, String)]
+
+      override def put(key: Object, value: Object) = {
+        val k = key.toString
+        val v = value.toString
+
+        parameters += k -> v
+        super.put(k, v)
+      }
+    }
+
+    try {
+      properties.load(stm)
+      properties.parameters.toList
+    } finally {
+      stm.close()
     }
   }
+
+  private val truthy = Set("y", "yes", "true")
+  private val falsy = Set("n", "no", "false")
```

Comment:
`$if(__show)$now you see me$endif$`

`show` would be a regular string field with value, say, `"yes"`. it could not be used directly as a conditional.

i like this alternative because it is guaranteed to not break existing templates: i can check whether `__field` already exists in the properties file and not override it.

of course, the `__` prefix is just a suggestion; we could use `_show_`, `__show__`, `show?`:

```
$if(_show_)$now you see me$endif$

$if(__show__)$now you see me$endif$

$if(show?)$now you see me$endif$
```

in the properties file it would still be just `show=yes`